frame/alliance/src/lib.rs

```diff
@@ -0,0 +1,1130 @@
+//!
+//! - Fellow: An account who is elevated from Ally by Founders and other Fellows from Ally.
+//!
+//! - Ally: An account who is approved by Founders and Fellows from Candidate. An Ally doesn't have
+//!   voting rights.
+//!
+//! - Candidate: An account who is trying to become a member. The applicant should already have an
+//!   approved identity with website. The application should be submitted by the account itself with
+//!   some token as deposit, or be nominated by an existing Founder or Fellow for free.
+//!
+//! - Blacklist: A list of bad websites and addresses, and can be added or removed items by Founders
+//!   and Fellows.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Functions
+//!
+//! #### For General Users
+//! - `submit_candidacy` - Submit the application to become a candidate with deposit.
+//!
+//! #### For Members (All)
+//! - `retire` - Member retire to out of the Alliance and release its deposit.
+//!
+//! #### For Members (Founders/Fellows)
+//!
+//! - `propose` - Propose a motion.
+//! - `vote` - Vote on a motion.
+//! - `close` - Close a motion with enough votes or expired.
+//! - `set_rule` - Initialize or update the alliance's rule by IPFS hash.
+//! - `announce` - Make announcement by IPFS hash.
+//! - `nominate_candidacy` - Nominate a non-member to become a Candidate for free.
+//! - `approve_candidate` - Approve a candidate to become an Ally.
+//! - `reject_candidate` - Reject a candidate and slash its deposit.
+//! - `elevate_ally` - Approve an ally to become a Fellow.
+//! - `kick_member` - Kick a member and slash its deposit.
+//! - `add_blacklist` - Add some items of account and website in the blacklist.
+//! - `remove_blacklist` - Remove some items of account and website from the blacklist.
+//!
+//! #### For Members (Only Founders)
+//! - `veto` - Veto on a motion about `set_rule` and `elevate_ally`.
+//!
+//! #### For Super Users
+//! - `init_founders` - Initialize the founding members.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+mod types;
+pub mod weights;
+
+use frame_support::pallet_prelude::*;
+use frame_system::pallet_prelude::*;
+use sp_runtime::{
+	traits::{StaticLookup, Zero},
+	RuntimeDebug,
+};
+use sp_std::prelude::*;
+
+use frame_support::{
+	codec::{Decode, Encode, MaxEncodedLen},
+	dispatch::{
+		DispatchError, DispatchResult, DispatchResultWithPostInfo, Dispatchable, GetDispatchInfo,
+		PostDispatchInfo,
+	},
+	ensure,
+	scale_info::TypeInfo,
+	traits::{
+		ChangeMembers, Currency, Get, InitializeMembers, IsSubType, OnUnbalanced,
+		ReservableCurrency,
+	},
+	weights::{Pays, Weight},
+};
+
+pub use pallet::*;
+pub use types::*;
+pub use weights::*;
+
+/// Simple index type for proposal counting.
+pub type ProposalIndex = u32;
+
+type UrlOf<T, I> = BoundedVec<u8, <T as pallet::Config<I>>::MaxWebsiteUrlLength>;
```

Comment:
```suggestion
type UrlOf<T, I = ()> = BoundedVec<u8, <T as pallet::Config<I>>::MaxWebsiteUrlLength>;
```

Reply:
actually I may want to drop all the default argument for `I` because that will usually causing confusing compile issue if missed